crawlerr/services/crawler_service.py

```
@@ -442,6 +393,378 @@ async def get_sitemap_urls(self, base_url: str) -> List[str]:
+        filter_chain: FilterChain,
+        stats: CrawlStatistics
+    ) -> List[PageContent]:
+        """
+        Manual BFS crawling with proper URL filtering as fallback.
+        
+        This is used when Crawl4AI's native deep crawl doesn't return deep_crawl_results.
+        """
+        pages = []
+        visited_urls = {start_url}
+        url_queue = [(start_url, 0)]
+        domain = urlparse(start_url).netloc
```

Ruff (0.12.2)

Local variable `domain` is assigned to but never used

Remove assignment to unused variable `domain`

(F841)